src/services/codefixes/importFixes.ts

```diff
@@ -356,6 +357,24 @@ function createImportAdderWorker(sourceFile: SourceFile | FutureSourceFile, prog
         }
     }
 
+    function addImportForExternalModuleSymbol(symbol: Symbol, originalSymbol: Symbol, isValidTypeOnlyUseSite: boolean) {
+        const useRequire = shouldUseRequire(sourceFile, program);
+        if (originalSymbol.valueDeclaration) {
+            const moduleSpecifier = moduleSpecifiers.getLocalModuleSpecifierBetweenFileNames(
+                sourceFile,
+                originalSymbol.valueDeclaration.getSourceFile().fileName,
+                compilerOptions,
+                createModuleSpecifierResolutionHost(program, host),
+            );
+            const info: FixInfo = {
+                fix: { kind: ImportFixKind.AddNew, importKind: ImportKind.Namespace, addAsTypeOnly: isValidTypeOnlyUseSite ? AddAsTypeOnly.Allowed : AddAsTypeOnly.NotAllowed, useRequire, moduleSpecifierKind: undefined, moduleSpecifier },

```

Comment:
Since there may be multiple different forms of import that can reference a whole module, you were asking about whether it’s ok to unconditionally coerce them into a namespace import. That might be fine, but it seems like it would make sense for this function to take an optional `referenceImport` like `addImportFromExportedSymbol`, and copy its kind and type-only status where appropriate.